src/pybind/mgr/dashboard_v2/controllers/rbd.py

```diff
@@ -22,21 +36,34 @@ def _format_bitmask(features):
         >>> Rbd._format_bitmask(45)
         'deep-flatten, exclusive-lock, layering, object-map'
         """
-        RBD_FEATURES_NAME_MAPPING = {
-            rbd.RBD_FEATURE_LAYERING: "layering",
-            rbd.RBD_FEATURE_STRIPINGV2: "striping",
-            rbd.RBD_FEATURE_EXCLUSIVE_LOCK: "exclusive-lock",
-            rbd.RBD_FEATURE_OBJECT_MAP: "object-map",
-            rbd.RBD_FEATURE_FAST_DIFF: "fast-diff",
-            rbd.RBD_FEATURE_DEEP_FLATTEN: "deep-flatten",
-            rbd.RBD_FEATURE_JOURNALING: "journaling",
-            rbd.RBD_FEATURE_DATA_POOL: "data-pool",
-            rbd.RBD_FEATURE_OPERATIONS: "operations",
-        }
-        names = [val for key, val in RBD_FEATURES_NAME_MAPPING.items()
+        names = [val for key, val in Rbd.RBD_FEATURES_NAME_MAPPING.items()
                  if key & features == key]
         return ', '.join(sorted(names))
 
+    @staticmethod
+    def _format_features(features):
+        """
+        Converts the features list to bitmask:
+
+        >>> Rbd._format_features(['deep-flatten', 'exclusive-lock', 'layering', 'object-map'])
+        45
+
+        >>> Rbd._format_features(None) is None
+        True
+
+        >>> Rbd._format_features('not a list') is None
+        True
+        """
+        if not features or not isinstance(features, list):
+            logger.warning("Given RBD features list is empty or None.")
```

Comment:
Either you expect `_format_features` to get `None` as features, then you should not log this at all, as it is already tested. 

Or, I would not expect this funciton to silently returns None, and instead raises an exception.

Reply:
Okay, will do.